assessments/projects/volcano/self-assessment.md

```diff
@@ -0,0 +1,242 @@
+  - Handles and validates requests from users, ensuring they conform to defined schemas and security protocols.
+  - Involved Actors:
+    + Users
+    + Volcano API Server
+  - Security Checks and System Actions:
+    + Request Authentication: Authenticate users making requests.
+    + Request Validation: Validate requests against security protocols and schemas.
+    + Response Integrity: Ensure the integrity and security of the data in the responses.
+
+6. API Server Functions: Authentication and Authorization
+  - Verifies user identities and checks whether they have sufficient permissions to perform requested actions.
+  - Involved Actors:
+    + Users
+    + Volcano API Server
+  - Security Checks and System Actions:
+    + Identity Verification: Confirm the identity of users.
+    + Permission Checks: Verify user permissions for requested actions.
+
+7. Kubernetes Cluster Interactions: Resource Provisioning
+  - Kubernetes cluster provides the necessary resources (like CPU, memory, and storage) as dictated by Volcano's scheduling decisions.
+  - Involved Actors:
+    + Kubernetes Cluster
+    + Volcano Scheduler
+  - Security Checks and System Actions:
+    + Resource Allocation Security: Ensure resources are provided securely and in compliance with policies.
+    + Access Control Enforcement: Kubernetes enforces access controls on resource provision.
+
+
+8. Kubernetes Cluster Interactions: Enforcement of Security Policies
+  - Kubernetes enforces security policies set by administrators, such as network policies and access controls, affecting how jobs are run and isolated.
+  - Involved Actors:
+    + Kubernetes Cluster
+  - Security Checks and System Actions:
+    + Policy Implementation: Enforce network and access control policies.
+    + Security Compliance Monitoring: Continuously monitor for compliance with set security policies.
+
+9. Integration with External Systems
+  - Data Exchange: Volcano may exchange data with integrated systems (like TensorFlow, PyTorch) for processing jobs. This includes sending job data and receiving results.
+  - Involved Actors
+    + Volcano
+    + Integrated Systems (e.g., TensorFlow, PyTorch)
+  - Security Checks and System Actions
+    + Secure Data Exchange: Ensure data exchanged with external systems is secure and complies with protocols.
+    + Compliance Check: Verify interactions adhere to established security standard
+
+### Goals
+
+- Efficient, optimized batch job scheduling in Kubernetes.
+- Secure handling of jobs and data within Kubernetes.
+
+
+### Non-goals
+
+- Replacing core Kubernetes functionalities.
+- Non-batch processing tasks (such as data-set processing/analysis, or machine learning).
+
+
+## Self-assessment use
+
+This self-assessment is created by the Volcano team to perform an internal analysis of the project's security. It is not intended to provide a security audit of Volcano, or function as an independent assessment or attestation of Volcano's security health.
```

Comment:
Is this true? This effort is independent of the team, from my knowledge, unless there was some collaboration I missed on this self-assessment.

Reply:
It was initially intended to be a collaboration between the Volcano team and the students, but we were unable to get any feedback from the team.
Fixed it to reflect the independence from the official team.

Reply:
Ah okay, that makes more sense